# HG changeset patch
# User Olaf Till <olaf.till@uni-jena.de>
# Date 1292513944 28800
#      Thu Dec 16 07:39:04 2010 -0800
# Node ID c767bb1afa03632932edc05379d585cc54f4669e
# Parent  690181867e54551fe7ff06a1b76c35b68001a55c
sqp.m: Fix indexing error in sqp bounds selection

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2010-12-16  Olaf Till <olaf.till@uni-jena.de>
+
+	* optimization/sqp.m: Fix indexing error in sqp bounds selection. 
+
 2010-12-14  Doug Stewart <doug.dastew@gmail.com>
 
 	* plot/bar.m: Improve the docstring.
 
 2010-12-14  John W. Eaton  <jwe@octave.org>
 
 	* plot/private/__quiver__.m: Use high contrast color black for arrows
 	in quiver3 plots.  Bug #31802.
diff --git a/scripts/optimization/sqp.m b/scripts/optimization/sqp.m
--- a/scripts/optimization/sqp.m
+++ b/scripts/optimization/sqp.m
@@ -307,33 +307,33 @@ function [x, obj, info, iter, nf, lambda
     else
       ## constraint inequality function with bounds present
       global __sqp_lb__;
       lb_idx = ub_idx = true (size (x0));
       ub_grad = - (lb_grad = eye (rows (x0)));
       if (isvector (lb))
         __sqp_lb__ = tmp_lb = lb(:);
         lb_idx(:) = tmp_idx = (lb != -Inf);
-        __sqp_lb__ = __sqp_lb__(tmp_idx);
+        __sqp_lb__ = __sqp_lb__(tmp_idx, 1);
         lb_grad = lb_grad(lb_idx, :);
       elseif (isempty (lb))
         if (isa (x0, "single"))
           __sqp_lb__ = tmp_lb = -realmax ("single");
         else
           __sqp_lb__ = tmp_lb = -realmax;
         endif
       else
         error ("sqp: invalid lower bound");
       endif
 
       global __sqp_ub__;
       if (isvector (ub))
         __sqp_ub__ = tmp_ub = ub(:);
         ub_idx(:) = tmp_idx = (ub != Inf);
-        __sqp_ub__ = __sqp_ub__(tmp_idx);
+        __sqp_ub__ = __sqp_ub__(tmp_idx, 1);
         ub_grad = ub_grad(ub_idx, :);
       elseif (isempty (ub))
         if (isa (x0, "single"))
           __sqp_ub__ = tmp_ub = realmax ("single");
         else
           __sqp_ub__ = tmp_ub = realmax;
         endif
       else
